### PR TITLE
add ability to pass timestamp

### PR DESCRIPTION
**Pull request by gregory**

### CODE REVIEW

File: spec/src/indexSpec.js [hunk 1]
```diff
@@ -136,6 +136,41 @@ describe('cloudwatch-metrics', function() {
...
+        Value: 'PROD'
+      }], {
+        withTimestamp: true,
+        sendInterval: 1000, // mocha defaults to a 2 second timeout so setting
+        // larger than that will cause the test to fail if we
+        // hit the timeout
+        sendCallback: done,
+      });
+
+      metric.put(1, 'metricName', [{Name:'ExtraDimension',Value: 'Value'}]);
```
Comment (skeggse) [inline spec/src/indexSpec.js]:
```suggestion
      metric.put(1, 'metricName', [{Name: 'ExtraDimension', Value: 'Value'}]);
```

File: spec/src/indexSpec.js [hunk 2]
```diff
@@ -136,6 +136,41 @@ describe('cloudwatch-metrics', function() {
...
+            Dimensions: [{
+              Name: 'environment',
+              Value: 'PROD'
+            }, {
+              Name: 'ExtraDimension',
+              Value: 'Value'
+            }],
+            MetricName: 'metricName',
+            Unit: 'Count',
+            Timestamp: jasmine.any(Number),
```
Comment (skeggse) [inline spec/src/indexSpec.js]: Could also verify that it's a number that's `>=` the value of `Date.now()` when the test case started

**Comment by gregory** [comment]
@skeggse - addressed your feedbacks. 
As for the Travis setup duplicated in #24  - i thought those solve different issues. What about i rebase whichever PR get merged last?

**Comment by gregory** [comment]
@skeggse  let me know what you think about the PR :)

File: spec/src/indexSpec.js [hunk 3]
```diff
@@ -136,6 +136,42 @@ describe('cloudwatch-metrics', function() {
...
+              Value: 'Value'
+            }],
+            MetricName: 'metricName',
+            Unit: 'Count',
+            Timestamp: jasmine.any(String),
+            Value: 1
+          }],
+          Namespace: 'namespace'
+        });
+        expect(Date.parse(data.MetricData[0].Timestamp)).toBeLessThan(Date.now());
```
Comment (skeggse) [inline spec/src/indexSpec.js]: millisecond precision affords for this to be the same as the timestamp
```suggestion
        expect(Date.parse(data.MetricData[0].Timestamp)).toBeLessThanOrEqual(Date.now());
```

**Pull request merged**